apps/src/lib/kits/maker/boards/microBit/MicroBitComponents.js

```diff
@@ -70,6 +71,10 @@ export function cleanupMicroBitComponents(
       rangeMax: 255,
       currentReading: 0
     };
+    components.lightSensor.buffer = new Float32Array(
+      MAX_SENSOR_BUFFER / SAMPLE_RATE
+    );
+    components.lightSensor.bufferIndex = 0;
```

Comment:
How do these related to the similar buffer (and index) that are created in microBit/LightSensor.js?

Reply:
They are the same - is called when the program is reset (via run/reset). So this is an opportunity for us to refresh the state of the component - new range, new 'current data', etc. 

Originally, I figured I could skip this step - since this is a circular buffer, we'll just keep looping over old data and don't need to 'zero it out' between runs. However, then I thought about a case where a student requests 3 seconds of data after only running their program for one second. In that case, if we've not 'zero'ed' out between runs, they might get data from their old previous runs of the program - which seems like a data bleed we don't want. But if we 'zero out' between runs, they'd get the value zero for the array values that are yet to be written, which feels like it would be more logical for students.

Reply:
Ah, that makes sense.  Agreed that resetting this data between runs is a good idea.

I'd suggest factoring the common code into a helper function in LightSensor.js, and which this code can then call.  That way, we don't have duplicated code, and this code doesn't have to "reach into" the internal data of lightSensor.

I'd also suggest only doing a `new` for the initial allocation of the buffer, and then resetting its contents to 0, perhaps using the new `.fill(0)` function.  I'm not sure of performance characteristics, but this feels better behaviour than making an additional memory allocation and relying on the garbage collector to clean things up.

Separately, if we are including 0s in the average, they will drag the value down unnecessarily, won't they?  It seems it would be better to average only the values we have written.  In that case, we would need to know how many values we have actually stored during the current run.  And in that case, we probably don't need to 0 anything out, we just need to track which array entries have valid values for the current run.

Reply:
To your last point - yes - I like it! I'm planning to implement it such that, if the user requests the average over 6 indices, but we've only recorded 3 indices, we'll return the average just over the 3 indices, rather than throwing an error or returning some other value.